core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
+                    $fields = array_keys( $this->modx->getFields($elements[$element]['class_key']));
                 }
+                
+                foreach($fields as $key => $value){
+                    $activeClass = $value == $query[2] ? 'active ' : '';
+                    $options .= '<li><a data-value="'.$value.'" class="'.$activeClass.$btn.'">'.$value.'</a></li>';
+                }                
+                $result['options'] = $options;
+                
+                $this->results[] = $result;
+            }
+            else{
+                $this->results[] = $validate;
             }
         }
-
-        return $this->outputArray($this->results);
+        
+        
+    /* :r:pagetitle: */
+        if($count == 4){
+            $validate = validate_Uberbar($elements,$query);
+            
+            if($validate === true)
+            {
+                $result['header'] = $this->modx->lexicon('uberbar_enter_searchstring_header');
+                $result['msg'] =    $this->modx->lexicon('uberbar_enter_searchstring_msg');
+                $result['options'] = '';
+                
+                $this->results[] = $result;
+            }
+            else{
+                $this->results[] = $validate;
+            }            
+        }
+             
+    /* :r:pagetitle:*test*: */
+        if($count == 5){
+            $validate = validate_Uberbar($elements,$query);
+
+            if($validate === true){
+                // Process search query, paginate on demand
+                if(!empty($elements[$element]['class_key'])){
+                    $this->queryElement($query,$elements,$uberbar_mode);
+                }
+                elseif($element == 'modx' && $field == 'config'){
+                    // 2do
+                }
+                else{
+                    return;
+                }
+                
+            }
+            else{
+                $this->results[] = $validate;
+            }            
+        }        
+
+        if($count > 5){
+            if($validate === true) {
+                $validate = validate_Uberbar($elements,$query,$uberbar_mode);
+            }
+            else{
+                $this->results[] = $validate;
+            }            
+        }
+        
+        
     }
+    
+    public function queryElement($query,$elements,$uberbar_mode){
+        
+        $startTime = microtime(true);
+        
+        $maxResults = $this->maxResults;
+        
+        $btn = "x-btn x-btn-small x-btn-icon-small-left primary-button x-btn-noicon";
+        
+        $element = $query[1];
+        $field   = $query[2];
+        $search  = $query[3];
+        
+        $class_key = $elements[$element]['class_key'];
+        $fields = $elements[$element]['fields'];
+        
+        /*
+        if($element == 'modx' AND $field == 'config'){
+            // 2do lookup system settings
+        }
+        else {
+            
+        }
+        */
```

Comment:
as long as we don't have the ability to "open" system settings with a certain item, I don't think this will be very useful. 

manager/?a=system/settings&key=site_url

Any idea how difficult it would be to add that feature?